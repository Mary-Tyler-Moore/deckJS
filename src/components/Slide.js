import React from 'react'
import { SlideTitle } from './'
import { Container, Hero, HeroBody } from 'bloomer'

const Slide = ({ title, subtitle, color, content }) => (
  <Hero isFullHeight isColor={color} key={title}>
    <HeroBody style={{ display: 'flex', flexDirection: 'column' }}>
      <SlideTitle title={title} subtitle={subtitle} />
      <Container hasTextAlign="centered">{content()}</Container>
    </HeroBody>
  </Hero>
)

export default Slide
