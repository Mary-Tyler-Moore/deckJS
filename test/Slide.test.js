import React from 'react'
import Slide from '../src/components/Slide'
import { shallow, mount } from 'enzyme'

describe('Slide', () => {
  const mock = {
    title: 'Test',
    subtitle: 'Substitle',
    color: 'color',
    content: () => {}
  }

  it('should render', () => {
    const wrapper = shallow(<Slide {...mock} />)
    expect(wrapper).toMatchSnapshot()
  })
})