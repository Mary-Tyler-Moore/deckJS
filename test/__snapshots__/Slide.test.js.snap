// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slide
    color="color"
    content={[Function]}
    subtitle="Substitle"
    title="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "Test",
    "nodeType": "function",
    "props": Object {
      "children": <withHelpersModifiers(HeroBody)
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <SlideTitle
          subtitle="Substitle"
          title="Test"
        />
        <withHelpersModifiers(Container)
          hasTextAlign="centered"
        />
      </withHelpersModifiers(HeroBody)>,
      "isColor": "color",
      "isFullHeight": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SlideTitle
            subtitle="Substitle"
            title="Test"
          />,
          <withHelpersModifiers(Container)
            hasTextAlign="centered"
          />,
        ],
        "style": Object {
          "display": "flex",
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "subtitle": "Substitle",
            "title": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "hasTextAlign": "centered",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "Test",
      "nodeType": "function",
      "props": Object {
        "children": <withHelpersModifiers(HeroBody)
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <SlideTitle
            subtitle="Substitle"
            title="Test"
          />
          <withHelpersModifiers(Container)
            hasTextAlign="centered"
          />
        </withHelpersModifiers(HeroBody)>,
        "isColor": "color",
        "isFullHeight": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <SlideTitle
              subtitle="Substitle"
              title="Test"
            />,
            <withHelpersModifiers(Container)
              hasTextAlign="centered"
            />,
          ],
          "style": Object {
            "display": "flex",
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "subtitle": "Substitle",
              "title": "Test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "hasTextAlign": "centered",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
